kbase-extension/static/kbase/js/common/simpleLogViewer.js

```diff
@@ -0,0 +1,818 @@
+                    const newLines = lines
+                        .slice(renderedLines.length)
+                        .map((line) => this.renderLogLine(line))
+                        .join('\n');
+                    olElement.innerHTML += newLines;
+                } else {
+                    panel.innerHTML = t('ol')(
+                        {
+                            class: `${cssBaseClass}__log_line_container`,
+                            dataElement: LOG_LINES,
+                        },
+                        lines.map((line) => this.renderLogLine(line)).join('\n')
+                    );
+                }
+
+                // if we're autoscrolling, scroll to the bottom
+                if (this.state.scrollToEndOnNext) {
+                    this.doScrollToBottom();
+                }
+                this.renderButtons();
+            }
+
+            /**
+             * Update the log control buttons
+             * Also renders the widget state if debugging mode is on
+             */
+            renderButtons() {
+                const previousStatus = (this.lastJobState && this.lastJobState.status) || null;
+                let mode = this._statusToMode(previousStatus) || 'default';
+                if (mode === 'running' && this.state.scrollToEndOnNext) {
+                    mode = 'running_scrolling';
+                }
+                this._renderButtonState(mode);
+            }
+
+            _renderButtonState(mode) {
+                // Button state
+                const { buttons } = this.buttonsByJobState[mode];
+                // the log controls section may have been removed
+                // so use try/catch to suppress any errors
+                buttons.enabled.forEach((_button) => {
+                    try {
+                        this.ui.enableButton(_button);
+                    } catch (err) {
+                        // eslint-disable-next-line no-empty
+                    }
+                });
+                buttons.disabled.forEach((_button) => {
+                    try {
+                        this.ui.disableButton(_button);
+                    } catch (err) {
+                        // eslint-disable-next-line no-empty
+                    }
+                });
+            }
+        };
+
+    const ControllerMixin = (Base) =>
+        class extends Base {
+            /* Job requests */
+
+            /**
+             * Requests log lines starting from the last line
+             * in the stored logs
+             */
+            requestJobLog() {
+                this.ui.showElement('spinner');
+                this.state.awaitingLog = true;
+                const requestParams = {
+                    [jcm.PARAM.JOB_ID]: this.jobId,
+                };
+                const lastLine = this.model.getItem('lastLine');
+                if (lastLine) {
+                    requestParams.first_line = lastLine;
+                }
```

Comment:
new -- when requesting job logs, the simple log viewer checks the model to see what the last stored line is and requests logs from that line onwards, rather than requesting all logs every time.